test-extender.sh

```diff
@@ -13,14 +17,28 @@ FROM cnbs/sample-builder:bionic
 COPY ./lifecycle /cnb/lifecycle
 EOF
 
-docker build -t test-builder .
+docker build -t $REGISTRY_HOST/test-builder .
+docker push $REGISTRY_HOST/test-builder
+
+cat << EOF > Dockerfile.extender
+FROM scratch
+COPY ./lifecycle /cnb/lifecycle
+ENTRYPOINT /cnb/lifecycle/extender
+EOF
+
+docker build -f Dockerfile.extender -t $REGISTRY_HOST/extender .
+docker push $REGISTRY_HOST/extender
 
 cd $SAMPLES_REPO_PATH
 
 rm -rf $SAMPLES_REPO_PATH/kaniko
 mkdir -p $SAMPLES_REPO_PATH/kaniko
 rm -rf $SAMPLES_REPO_PATH/layers/kaniko
 mkdir -p $SAMPLES_REPO_PATH/layers/kaniko
+rm -rf $SAMPLES_REPO_PATH/kaniko-run
+mkdir -p $SAMPLES_REPO_PATH/kaniko-run
+rm -rf $SAMPLES_REPO_PATH/layers-run/kaniko
+mkdir -p $SAMPLES_REPO_PATH/layers-run/kaniko
```

Comment:
I guess a next step in this effort could be to try to make these directories useful for caching purposes (though alternative approaches to caching such as spinning up a registry process were also discussed).